sdk/servicebus/Microsoft.Azure.ServiceBus/src/Primitives/ConcurrentExpiringSet.cs

```diff
@@ -4,33 +4,45 @@
 namespace Microsoft.Azure.ServiceBus.Primitives
 {
     using System;
+    using System.Collections.Generic;
     using System.Collections.Concurrent;
+    using System.Threading;
     using System.Threading.Tasks;
 
     sealed class ConcurrentExpiringSet<TKey>
     {
         readonly ConcurrentDictionary<TKey, DateTime> dictionary;
+        readonly ICollection<KeyValuePair<TKey, DateTime>> dictionaryAsCollection;
         readonly object cleanupSynObject = new object();
+        CancellationTokenSource tokenSource = new CancellationTokenSource(); // doesn't need to be disposed because it doesn't own a timer
```

Comment:
I believe CancellationTokenSource implements IDisposable and I think it might internally use a timer.

Reply:
It does. As far as I remember it only allocates disposable resources like timers when used with timeouts. Happy though to add

Reply:
See https://referencesource.microsoft.com/#mscorlib/system/threading/CancellationTokenSource.cs,554 that's why I did not implement the dispose call because we are not using the registrations or the timer path of the disposable